### DISCUSSION TITLE
Feature Release: 0.2

### DISCUSSION THREAD

**Discussion by cboxdoerfer** [cboxdoerfer/fsearch#381] [Announcements]
To get the bad news out of the way quickly: Unfortunately Snap support had to be removed. For more information about that see: <https://github.com/cboxdoerfer/fsearch/wiki/Snap-is-no-longer-officially-supported>

But now to the positive news...

This is and will probably be one of the most important releases for FSearch.

# Greatly improved search engine

The most important news is the updated search engine. Now it's finally possible to search for entries with a particular size, modification date, folder depth, extension, content type, number of children and more. For example, to find every MP4 file, larger than 1GB, you can simply type: 

`ext:mp4 size:>1gb`

Like before, empty space will be treated as an `AND` operator, but with this new release you can now also build queries with `OR` and `NOT` operators. For example, to find all JPG and PNG files, which were modified last month, with `mum` _or_ `dad` in their name, you can simply type:

`(mum OR dad) ext:jpg;png dm:lastmonth`

For a more detailed list of features of the updated search engine see: https://github.com/cboxdoerfer/fsearch/wiki/Search-syntax

# Custom filters

FSearch already had a few selected filters included, such as `Videos`, `Documents` or `Archives`. Now it's possible to add new filters in the preferences dialog and edit the default ones. This feature is especially useful for frequently used queries. For example, if you often find yourself searching within your backup drive you can now add a new filter:

![Screenshot from 2022-08-15 12-57-21](https://user-images.githubusercontent.com/6108388/184623582-c67728b7-df67-43fe-94d7-e39529c9b053.png)

After that `Backup` will show up in the filter drop down menu, next to the search field, where you can quickly activate the filter. Or you can also use the macro name, followed by an `:`, in the search field, to activate filters on demand. For example, to find all files in your backup which were last modified more than 3 years ago you can then type:

`backup: dm:<past3years`

# Noteworthy changes

* New search engine
* Custom filters
* Greatly improved highlighting of search terms
* Slow queries can now be aborted
* Sorting can now be aborted
* Improved sort performance
* Improved scroll performance
* Remember file selection after a database update
* Added filter for desktop applications
* Option to launch desktop applications (`.desktop` files)
* Show desktop application icons for `.desktop` files
* Option to exit FSearch when **`ESC`** is pressed (@mrubli)
* Scroll the result list when selecting files with the rubber band if necessary
* Invert selection with rubber band when **`Ctrl`** is pressed
* `Open with` menu only shows applications which support all selected files
* Improved opening of multiple files with the same type (they get now passed as a list to applications, instead of one by one)
* Better support for opening files in sandboxed mode
* Add menu items with  links to the issue tracker, forum and donation pages
* Exclude `/proc` and `/sys` by default
* Open context menu when **`Menu`** key or **`Shift + F10`** are pressed
* and tons of bug fixes, translation updates and smaller improvements

# What's next?

Now that searching is almost as powerful as I want it to be, the next 0.3 release will primarily focus on the database. My plan is to finally add the ability to monitor for filesystem changes, so you don't have to update the database manually so often. It will also add the ability to index (and search) additional information (like creation or access time).

# And last but not least: Tank you!

I'm really grateful for all the contributions, whether it were code changes, bug reports, translations or donations!

<hr /><em>This discussion was created from the release <a href='https://github.com/cboxdoerfer/fsearch/releases/tag/0.2'>Feature Release: 0.2</a>.</em>

**Comment by azlan** [comment]
Tank you sir

**Comment by sl0J0n** [comment]
Many tanks, SIR!
BTW, I have 0.1 installed; any update option?
Better to 'uninstall'? or 'autoremove'?, 'purge'?, or install overwrite w/ 0.2?
Thank you in advance for your kind assistance.
sl0j0n

**Comment by benyaminl** [comment]
Can the macro added to wiki?
